crates/ra_hir/src/ty/infer.rs

```diff
@@ -63,6 +63,24 @@ enum ExprOrPatId {
 
 impl_froms!(ExprOrPatId: ExprId, PatId);
 
+/// Binding modes inferred for patterns.
+/// https://doc.rust-lang.org/reference/patterns.html#binding-modes
+#[derive(Copy, Clone, Debug, Eq, PartialEq)]
+enum BindingMode {
+    Move,
```

Comment:
`rustc` calls this `BindByValue` (and takes a mutability argument).

I chose to call it `Move` to match the reference documentation, and the match ergonomics RFC which refer to it that way. I'm very happy to change it if consistency with `rustc` is more important.